api/v1/services/ProjectService.js

```diff
@@ -0,0 +1,208 @@
+var Checkit = require('checkit');
+var _Promise = require('bluebird');
+var _ = require('lodash');
+
+var Mentor = require('../models/Mentor');
+var Project = require('../models/Project');
+var ProjectMentor = require('../models/ProjectMentor');
+
+var errors = require('../errors');
+var utils = require('../utils');
+var roles = require('../utils/roles');
+
+const projectKeys = ["name", "description", "repo", "isPublished"];
+
+/**
+ * Checks to see if a requestor valid permissions to create a new project
+ * @param  {User} user creating the new project
+ * @return {Promise} resolving to true if the user is an organizer
+ * @throws InvalidParameterError when a user does not have correct permissions
+ */
+module.exports.canCreateProject = function (creator) {
+	if(creator.hasRole(roles.SUPERUSER) || creator.hasRole(roles.ORGANIZERS)){
+		return _Promise.resolve(true);
+	}
+
+	var message = "A project cannot be created with the provided credentials";
+	return _Promise.reject(new errors.UnauthorizedError(message));
+}
+
+
+/**
+ * Creates a project with the specificed attributes
+ * @param  {Object} Contains name, description, repo, and is_published
+ * @return {Promise} resolving to the newly-created project
+ * @throws InvalidParameterError when a project exists with the specified name
+ */
+module.exports.createProject = function (attributes) {
+	if(typeof attributes.repo === 'undefined'){
+		attributes.description = '';
+	}
+	if(typeof attributes.isPublished === 'undefined'){
```

Comment:
Since you have `lodash` imported already, just do `if (_.isUndefined(attributes.isPublished))` You might also consider checking `_.isNull`.